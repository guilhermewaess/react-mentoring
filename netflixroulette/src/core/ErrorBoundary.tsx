import * as React from 'react';

interface IState {
  hasError: boolean;
}

export default class ErrorBoundary extends React.Component {
  public state: IState;

  constructor(props: any) {
    super(props);
    this.state = { hasError: false };
  }

  public componentDidCatch(error: any, errorInfo: any) {
    this.setState({ hasError: true });
    // tslint:disable-next-line
    console.log('Ups!', error, errorInfo);
  }

  public render() {
    if (this.state.hasError) {
      return <h1>Ops</h1>;
    }

    return this.props.children;
  }
}
