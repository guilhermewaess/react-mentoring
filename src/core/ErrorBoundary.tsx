import * as React from 'react';

interface IState {
  hasError: boolean;
}

export default class ErrorBoundary extends React.Component<any, IState> {
  public state: IState;

  constructor(props: any) {
    super(props);
    this.state = { hasError: false };
  }

  public componentDidCatch(error: any, errorInfo: any) {
    this.setState({ hasError: true });
  }

  public render() {
    if (this.state.hasError) {
      return <h1>Ops</h1>;
    }

    return this.props.children;
  }
}
