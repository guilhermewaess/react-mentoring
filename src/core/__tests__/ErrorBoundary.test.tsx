import { mount, ReactWrapper } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import * as React from 'react';
import ErrorBoundary from './../ErrorBoundary';

describe('ErrorBoundary', () => {
  let component: ReactWrapper<ErrorBoundary>;
  beforeEach(() => {
    component = mount(<ErrorBoundary children={<div>Children</div>} />);
  });
  describe('when construct', () => {
    it('should have state', () => {
      expect(component.state()).toEqual({ hasError: false });
    });
    it('should render with state.hasError false', () => {
      expect(mountToJson(component)).toMatchSnapshot();
    });
  });
  // TODO: https://github.com/airbnb/enzyme/issues/1553
  // Test componentDidCatch in the future
  describe('when component catch an error', () => {
    beforeEach(() => {
      component.setState({ hasError: true});
    });
    it('should render with state.hasError true', () => {
      expect(mountToJson(component)).toMatchSnapshot();
    });
  });
});
