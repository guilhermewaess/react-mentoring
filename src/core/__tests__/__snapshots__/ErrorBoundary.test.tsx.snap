// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary when component catch an error should render with state.hasError true 1`] = `
<ErrorBoundary>
  <h1>
    Ops
  </h1>
</ErrorBoundary>
`;

exports[`ErrorBoundary when construct should render with state.hasError false 1`] = `
<ErrorBoundary>
  <div>
    Children
  </div>
</ErrorBoundary>
`;
